specs/core/0_fork-choice.md

```diff
@@ -248,10 +248,17 @@ def on_attestation(store: Store, attestation: Attestation) -> None:
     # Cannot calculate the current shuffling if have not seen the target
     assert target.root in store.blocks
 
+    # Attestations target be for a known block. If target block is unknown, delay consideration until the block is found
```

Comment:
> delay consideration until the block is found

Considered returning the block root to await instead. But then there are slot/time based exit conditions as well, and we likely don't want more complicated scheduling in the spec. (Also since not all cases may be fulfilled later, e.g. a non-existing block root).

Agree that there is some freedom in processing here, and the delayed consideration shouldn't be part of the spec.
Same for delayed-consideration cases below.

A warning at the top of the function may be in place: "An `Attestation` that is asserted as invalid may be valid at a later time, consider scheduling it for later processing in such case"